### PR TITLE
updated emscripten to version 1.36.14, fixes #6771

### LINKED ISSUE #6771: emscripten formula has out of date dependencies
```sh
 emcc -v
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.36.13
clang version 3.9.0  (emscripten 1.36.5 : 1.36.5)
Target: x86_64-apple-darwin16.1.0
Thread model: posix
InstalledDir: /usr/local/opt/emscripten/libexec/llvm/bin
ERROR:root:Emscripten, llvm and clang build versions do not match, this is dangerous (1.36.13, 1.36.5, 1.36.5)
ERROR:root:Make sure to rebuild llvm and clang after updating repos
INFO:root:(Emscripten: Running sanity checks)
```

See [here](https://github.com/kripken/emscripten/issues/4710#issuecomment-259558918) for more.

### PR DESCRIPTION
**Pull request by bodokaiser**
Did not test `brew install --HEAD emscripten` but it should now working using the `master` branch over `incoming` (`incoming` failed actually).

Normal install works now. Problem with `cmake` was a missing `depends_on "cmake" => ":build"`.

### CODE REVIEW

**Comment by bodokaiser** [comment]
Please wait with merge. I may add another dependency. See https://github.com/kripken/emscripten/issues/4710

File: Formula/emscripten.rb [hunk 1]
```diff
@@ -3,18 +3,21 @@ class Emscripten < Formula
   homepage "https://kripken.github.io/emscripten-site/"
 
   stable do
-    url "https://github.com/kripken/emscripten/archive/1.36.13.tar.gz"
-    sha256 "5e5370384a80775eaed63f38dcbc11f38620239f21c3fd6ba3725b5c982a2736"
+    version "1.36.14"
+    url "https://github.com/kripken/emscripten/archive/#{version}.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/emscripten.rb]: This should stay as `1.36.14`

File: Formula/emscripten.rb [hunk 2]
```diff
@@ -3,18 +3,21 @@ class Emscripten < Formula
...
+    sha256 "89febe6c56c36ded3a6323d40342196d961eb1a7878b32912a649734962cb5ee"
 
     resource "fastcomp" do
-      url "https://github.com/kripken/emscripten-fastcomp/archive/1.36.5.tar.gz"
-      sha256 "322501d14eb90b5590d463ef2ae1b358c07c590440d7bd21b60ea88885bc2fa0"
+      url "https://github.com/kripken/emscripten-fastcomp/archive/#{version}.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/emscripten.rb]: Sorry, I forgot this doesn't work as expected. You can either take the approach from `swift.rb` or (preferred) write something in the `test do` that will reproduce the failure you were seeing when the versions were mismatched. That will stop this being merged with that error.

File: Formula/emscripten.rb [hunk 3]
```diff
@@ -3,18 +3,21 @@ class Emscripten < Formula
...
     end
 
     resource "fastcomp-clang" do
-      url "https://github.com/kripken/emscripten-fastcomp-clang/archive/1.36.5.tar.gz"
-      sha256 "b6a35fe26efaaaaea5d3d1139e61d5754760f03bed0a4af87236767d1a56b00d"
+      url "https://github.com/kripken/emscripten-fastcomp-clang/archive/#{version}.tar.gz"
+      sha256 "d33574f378acde198a2407a88cfa2725d8853dee535f982ec5fac92b4180f3aa"
     end
+
+    depends_on "cmake" => :build
```
Comment (MikeMcQuaid) [inline Formula/emscripten.rb]: This can be removed, it's repeated below.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/emscripten.rb [hunk 4]
```diff
@@ -3,17 +3,18 @@ class Emscripten < Formula
   homepage "https://kripken.github.io/emscripten-site/"
 
   stable do
-    url "https://github.com/kripken/emscripten/archive/1.36.13.tar.gz"
-    sha256 "5e5370384a80775eaed63f38dcbc11f38620239f21c3fd6ba3725b5c982a2736"
+    url "https://github.com/kripken/emscripten/archive/1.36.14.tar.gz"
+    sha256 "89febe6c56c36ded3a6323d40342196d961eb1a7878b32912a649734962cb5ee"
 
+    emscripten_tag = "#{version}.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/emscripten.rb]: The `.tar.gz` shouldn't be here.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `emscripten 1.36.14`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request closed without merging**